### DISCUSSION TITLE
v7 `primaryColor` inheritance solution

### DISCUSSION THREAD

**Discussion by alexsasharegan** [mantinedev/mantine#4798] [Q&A] [answered]
In version 6.x I was able to nest various sections of my app with a simple `MantineProvider` that modified the `primaryColor`. This had the desired effect of creating a branded look for various distinct sections of the app. 

Now that the provider model has changed and a single provider is the proper way to use this, how might I go about migrating this? The most apparent solution to me would be to update a CSS variable that I would imagine to be named something like this: `--mantine-primary-color`, however I have found no documentation addressing this kind of problem.

Any ideas on how I might implement this behavior in v7? Much appreciated!

**Comment by rtivital** [comment, marked as answer]
You can still use nested MantineProviders for primary color, see Mantine UI as reference – https://github.com/mantinedev/ui.mantine.dev/blob/master/components/ComponentCanvas/ComponentCanvas.tsx

**Reply by alexsasharegan** [reply]
Interesting. I see you are still using the `inherit` prop, but for those of us using Typescript, this prop shows itself as being removed, and when I look into the code shipped in v7, that prop is ignored. This just makes me concerned I need to mix in all the root theme config in any of these nested Providers.
